### PR TITLE
Migrate to modular flyway (v10) + add postgres module

### LINKED ISSUE #12247: Investigate flyway version upgrade issue
### Description/Use Case
Dependabot #12223 is causing integration tests to fail. Investigate the underlying cause and fix it.

### Risks/Impacts/Considerations
In any resulting test changes be careful to consider any similar changes needed to actual code.

### Dev Notes
The new version of `flyway` introduces modular db support and the postgres plugin is now missing.

### Acceptance Criteria
- Upgraded to latest version of `flyway` with no test issues
- Any new flyway-related dependencies are added to the appropriate dependabot group

### PR DESCRIPTION
**Pull request by thetaurean**
This PR bumps `flyway` from `9.3` to `10.0.1`, adding a dependency for the newly introduced `flyway` `postgresql module`. Version 10 of flyway introduced modular support for databases. These dependencies need to be injected on the build classpath (as opposed to the project classpath).

Test Steps:
1. Gradle sync
2. Run test suite

## Changes
- Bump `flyway` to `10.0.1`
- Add modular dependency for `flyway-database-postgresql` to `buildscript`

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?

### Process
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #12247

### CODE REVIEW

**Review by mkalish — APPROVED** [review]
Couple of small comments, but looks good!

I wiped my DB, rebuilt and ran the smoketests

**Comment by mkalish** [comment]
@thetaurean We should keep an eye on the automatic deploy to staging when this gets merged

File: prime-router/src/main/kotlin/azure/DatabaseAccess.kt [hunk 1]
```diff
@@ -1444,10 +1445,12 @@ class DatabaseAccess(val create: DSLContext) : Logging {
             config.maxLifetime = 180000
             val dataSource = HikariDataSource(config)
 
+            val flywayConfig = Flyway.configure()
+            val pgConfig = flywayConfig.pluginRegister.getPlugin(PostgreSQLConfigurationExtension::class.java)
             // This setting makes flyway fall back to session locks as concurrent index creation cannot be done
             // within a transaction. This setting is needed as of flyway 9.19.4.
-            val flyway = Flyway.configure().configuration(mapOf(Pair("flyway.postgresql.transactional.lock", "false")))
-                .dataSource(dataSource).load()
+            pgConfig.setTransactionalLock(false)
```
[the remark this answers is not in the document]
Comment (thetaurean) [reply prime-router/src/main/kotlin/azure/DatabaseAccess.kt:1452]: No, that value must remain or we fail elsewhere (tests IIRC).

File: prime-router/build.gradle.kts [hunk 2]
```diff
@@ -780,6 +780,7 @@ buildscript {
         // Now force the gradle build script to get the proper library for com.nimbusds:oauth2-oidc-sdk:9.15.  This
         // will need to be removed once this issue is resolved in Maven.
         classpath("net.minidev:json-smart:2.5.0")
+        classpath("org.flywaydb:flyway-database-postgresql:10.0.1")
```
[the remark this answers is not in the document]
Comment (thetaurean) [reply prime-router/build.gradle.kts:784]: Good idea 👍 added ✅

**Review by arnejduranovic — APPROVED** [review]

**Pull request merged**